cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs

```diff
@@ -747,13 +746,13 @@ pKESVerificationKeyFile =
       <> Opt.help "Filepath of the hot KES verification key."
       )
 
-pMaybeProtocolMagicId :: Parser (Maybe Byron.ProtocolMagicId)
-pMaybeProtocolMagicId =
-  optional $ Byron.ProtocolMagicId <$>
+pNetwork :: Parser Network
```

Comment:
I am pretty sure you need `Parser (Maybe x)` for optparse to correctly detect that as an optional CLI parameter.

Reply:
oops, I forgot so supply the default value of Mainnet.